src/main/java/com/gmail/goosius/siegewar/objects/SiegeCamp.java

```diff
@@ -125,11 +125,16 @@ public void setAttackerPoints(int attackerPoints) {
 	 * Starts the Siege after the success of the SiegeCamp.
 	 */
 	public void startSiege() {
-		
-		// Retest that the nation can still pay the warchest.
-		if (!siegeType.equals(SiegeType.REVOLT) && TownyEconomyHandler.isActive() && !attacker.getAccount().canPayFromHoldings(SiegeWarMoneyUtil.calculateSiegeCost(targetTown))) {
-			TownyMessaging.sendPrefixedNationMessage((Nation)attacker, Translatable.of("msg_err_no_money"));
-			return;
+		//Test that the siege starter can pay the siege start cost
+		if (TownyEconomyHandler.isActive()) {
+			double siegeStartCost = SiegeWarMoneyUtil.calculateTotalSiegeStartCost(targetTown);
+			if (siegeType.equals(SiegeType.CONQUEST) && !attacker.getAccount().canPayFromHoldings(siegeStartCost)) {
+				TownyMessaging.sendPrefixedNationMessage((Nation)attacker, Translatable.of("msg_err_no_money"));
+				return;
+			} else if (siegeType.equals(SiegeType.REVOLT) && !targetTown.getAccount().canPayFromHoldings(siegeStartCost)) {
+				TownyMessaging.sendPrefixedTownMessage(targetTown, Translatable.of("msg_err_no_money"));
+				return;

```

Comment:
There was a recent addition that told the users how much money they needed to start a siege, because if they could not afford it, the were also not told how much they needed. These error messages should also explain how much they need to start a siege.

Reply:
It is below in the SiegeWarMoneyUtil:

`Translatable.of("msg_err_your_nation_cannot_afford_to_siege_for_x", TownyEconomyHandler.getFormattedBalance(cost))`

I don't know if you'd want to reuse the language string or have a new one.

Reply:
Comment addressed